tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,794 @@
+    # of what we asked for: if this is violated then
+    # it is something wrong with the client or test environment.
+    producer_tolerance = 2
+
+    @property
+    def expected_producer_duration(self) -> int:
+        return (self.write_bytes //
+                self.produce_byte_rate) * self.producer_tolerance
+
+    # consumers should receive at least half of the messages
+    # aligned with bitrate
+    consumer_tolerance = 2
+
+    @property
+    def max_consumer_duration(self) -> int:
+        return self.target_runtime * self.consumer_tolerance * \
+               self.total_iterations
+
+    # If we limit consume rate, then set message count
+    # consumer_message_count = msg_count // (
+    #     produce_byte_rate // consume_byte_rate) // consumer_tolerance
+    consumer_message_count = None
+
+    # Additional configs
+    retention_ms = -1
+    cloud_storage_spillover_manifest_max_segments = 10
+    local_retention_bytes = str(msg_size)
+
+
+class InfiniteRetentionTest(PreallocNodesTest):
+    DEFAULT_PARAMS = InfiniteRetentionParameters()
+
+    def __init__(self, test_context, *args, **kwargs):
+        self.params = self.DEFAULT_PARAMS
+
+        self.si_settings = SISettings(
+            test_context=test_context,
+            log_segment_size=self.params.segment_size,
+        )
+        kwargs['si_settings'] = self.si_settings
+
+        # Use interval uploads so that at end of test we may do an "everything
+        # was uploaded" success condition.
+        kwargs['extra_rp_conf'] = {
+            # We do not intend to do interval-triggered uploads during produce,
+            # but we set this property so that at the end of the test we may
+            # do a simple "everything was uploaded" check after the interval.
+            'cloud_storage_segment_max_upload_interval_sec':
+            self.params.segment_upload_interval,
+            # The test will assert that the number of manifest uploads does
+            # not exceed what we would expect based on this interval.
+            'cloud_storage_manifest_max_upload_interval_sec':
+            self.params.manifest_upload_interval,
+            # Default retention is infinite
+            'delete_retention_ms':
+            self.params.retention_ms,
+            # This causes spillover to hapen more often
+            'cloud_storage_spillover_manifest_max_segments':
+            self.params.cloud_storage_spillover_manifest_max_segments,
+        }
+        super().__init__(test_context, node_prealloc_count=2, *args, **kwargs)
+
+        self.rpk = RpkTool(self.redpanda)
+
+        self.topics = (TopicSpec(
+            retention_ms=self.params.retention_ms,
+            replication_factor=3,
+            partition_count=self.params.partition_count), )
+
+    def setUp(self):
+        # ensure nodes are clean before the test
+        super().setUp()
+
+    def tearDown(self):
+        # stop everything
+        for node in self.redpanda.nodes:
```

Comment:
What happens if you don't do this?  Usually tests do not implement `tearDown`, because duckape already stops all services (i.e. it is already calling stop_node for each node of redpanda, etc)

Reply:
Sometimes, when I debug on ducktape leader node, preallocated nodes got stuck when I kill the process, this offers freeing those nodes on a separate subsequent run. Will remove in final version probably.